shanoir-ng-datasets/src/main/java/org/shanoir/ng/dicom/DicomProcessing.java

```diff
@@ -41,4 +47,40 @@ public Attributes getDicomObjectAttributes(DatasetFile image, Boolean isEnhanced
 		}
 	}
 
+    public ExaminationAttributes getDicomExaminationAttributes(Study study, Boolean isEnhanced) throws PacsException {
+		ExaminationAttributes attributes = new ExaminationAttributes();
+		if (study != null) {
+			for (Serie serie : study.getSeries()) {
+				attributes.addAcquisitionAttributes(serie.hashCode(), getDicomAcquisitionAttributes(serie, isEnhanced));
```

Comment:
Hi @julien-louis, I am wondering, you are using hashCode here, instead of seriesInstanceUIDs, here but later you search with long ids for datasets, what is quite confusing. With kind regards, Michael

Reply:
I think that in the case of an import I don't have the dataset ids yet, so I went with this. Do you think it's too unreadable ?

Reply:
Actually there may be a big problem here. I'm addressing it.

Reply:
Ok I used seriesInstanceUIDs and SOPInstanceUID. Those are strings so I had to parametrize the two classes ExaminationAttributes and AcquisitionAttributes which is better because it reduces the risk of mixing up the two cases (db ids or dicom uids).